os/test/src/PathTests.scala

```diff
@@ -425,5 +425,10 @@ object PathTests extends TestSuite{
         }
       }
     }
+    test("issue159"){
+      val result = os.rel / Seq(os.up, os.rel/"hello", os.rel/"world")
+      val expected = os.up / "hello" / "world"
+      assert(result == expected)
```

Comment:
Would be nice to also cover the `ArrayPathChunk` case here, as it's also affected by #159.